src/gt4py/next/common.py

```diff
@@ -74,7 +84,7 @@ def __str__(self):
 
 
 @dataclasses.dataclass(frozen=True)
-class UnitRange(Sequence[int], Set[int]):
+class UnitRange(Sequence[int]):
```

Comment:
Because `UnitRange` is overriding the `__and__` method and providing a custom implementation for that expecting that the type of `other` is `UnitRange`. If I inherit from `Set[int]` I get the following mypy error:

```
src/gt4py/next/common.py:130:23: error: Argument 1 of "__and__" is incompatible with supertype "set"; supertype defines the argument type as "AbstractSet[object]"  [override]
src/gt4py/next/common.py:130:23: note: This violates the Liskov substitution principle
src/gt4py/next/common.py:130:23: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/gt4py/next/common.py:130:23: error: Argument 1 of "__and__" is incompatible with supertype "AbstractSet"; supertype defines the argument type as "AbstractSet[Any]"  [override]
``` 

If we inherit from set the `__and__` method expects that `other` has type `AbstractSet` which is not the case for us. It seems to me that UnitRange doesn't need to fully implement the Set interface, as it is mainly meant to represent a numerical range and doesn't behave like a full set.